business/authentication/auth_controller.go

```diff
@@ -0,0 +1,47 @@
+package authentication
+
+import (
+	"net/http"
+
+	"github.com/kiali/kiali/config"
+)
+
+// AuthController is the interface that all Kiali authentication strategies should implement.
+// An authentication controller is initialized during Kiali startup.
+type AuthController interface {
+	// Authenticate handles an HTTP request that contains credentials. The method to pass the credentials
+	// is chosen by the authentication controller implementation. The credentials are verified and if
+	// it is supported by the controller, RBAC permissions are verified to ensure that the logging in user
+	// has enough privileges to login to Kiali.
+	// An AuthenticationFailureError is returned if the authentication request is rejected (unauthorized). Any
+	// other kind of error means that something unexpected happened.
+	Authenticate(r *http.Request, w http.ResponseWriter) (*UserSessionData, error)
+
+	// ValidateSession restores a session previously created by the Authenticate function. The validity of
+	// the restored should be verified as much as possible by the implementing controllers.
+	// If the session is still valid, a populated UserSessionData is returned. Otherwise, nil is returned.
+	ValidateSession(r *http.Request, w http.ResponseWriter) (*UserSessionData, error)
+
+	// TerminateSession performs the needed procedures to terminate an existing session. If there is no
+	// active session, nothing is performed. If there is some invalid session, it is cleared.
+	TerminateSession(r *http.Request, w http.ResponseWriter)
+}
+
+var authController AuthController
+
+// GetAuthController gets the authentication controller that is currently configured and handling
+// user sessions and any authentication related requests.
+func GetAuthController() AuthController {
+	return authController
+}
+
+// InitializeAuthenticationController initializes the authentication controller associated to the
+// given strategy and prepares it to control user sessions and handle authentication requests.
+// This should be called during Kiali startup, before starting to listen to HTTP requests.
+func InitializeAuthenticationController(strategy string) {
+	persistor := CookieSessionPersistor{}
```

Comment:
Changing strategies should also invalidate any existing cookie sessions in case a user were to try and send an old session id with the new auth strategy.

Reply:
Yes. Although that's done once a request arrives, because the browser is the storage (because of the cookies).
That's the intention of this line: https://github.com/kiali/kiali/pull/4596/files#diff-a6762b12addd1f51f23491d16825ae2746eff746dbba3e9528015d0fc0913d7aR243